app/components/selected_users/index.tsx

```diff
@@ -0,0 +1,240 @@
+const EXPOSED_CHIP_HEIGHT = 0.33 * USER_CHIP_HEIGHT;
+const MAX_CHIP_ROWS = 3;
+const SCROLL_PADDING_TOP = 20;
+const SCROLL_VIEW_MAX_HEIGHT = SCROLL_PADDING_TOP + (CHIP_HEIGHT_WITH_MARGIN * MAX_CHIP_ROWS) + EXPOSED_CHIP_HEIGHT;
+const TABLET_MARGIN_BOTTOM = 20;
+const TOAST_BOTTOM_MARGIN = 8;
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            backgroundColor: theme.centerChannelBg,
+            borderBottomWidth: 0,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.16),
+            borderTopLeftRadius: 12,
+            borderTopRightRadius: 12,
+            borderWidth: 1,
+            elevation: 4,
+            maxHeight: SCROLL_VIEW_MAX_HEIGHT + BUTTON_HEIGHT,
+            overflow: 'hidden',
+            paddingHorizontal: 20,
+            shadowColor: theme.centerChannelColor,
+            shadowOffset: {
+                width: 0,
+                height: 8,
+            },
+            shadowOpacity: 0.16,
+            shadowRadius: 24,
+        },
+        toast: {
+            backgroundColor: theme.centerChannelColor,
+        },
+        users: {
+            paddingTop: SCROLL_PADDING_TOP,
+            paddingBottom: 12,
+            flexDirection: 'row',
+            flexGrow: 1,
+            flexWrap: 'wrap',
+        },
+        message: {
+            color: changeOpacity(theme.centerChannelColor, 0.6),
+            fontSize: 12,
+            marginRight: 5,
+            marginTop: 10,
+            marginBottom: 2,
+        },
+    };
+});
+
+export default function SelectedUsers({
+    selectedIds,
+    teammateNameDisplay,
+    showToast = false,
+    toastIcon,
+    toastMessage,
+    onPress,
+    onRemove,
+    buttonIcon,
+    buttonText,
+}: Props) {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const isTablet = useIsTablet();
+    const keyboardHeight = useKeyboardHeight();
+    const insets = useSafeAreaInsets();
+
+    const scrollViewHeight = useSharedValue(0);
+    const [isVisible, setIsVisible] = useState(false);
+
+    useEffect(() => {
+        setIsVisible(Boolean(Object.keys(selectedIds).length));
+    }, [selectedIds]);
+
+    const handleOnPress = async () => {
+        onPress();
+    };
+
+    const users = useMemo(() => {
+        const u = [];
+        for (const id of Object.keys(selectedIds)) {
+            if (!selectedIds[id]) {
+                continue;
+            }
+
+            u.push(
+                <SelectedUser
+                    key={id}
+                    user={selectedIds[id]}
+                    teammateNameDisplay={teammateNameDisplay}
+                    onRemove={onRemove}
+                    testID='create_direct_message.selected_user'
+                />,
+            );
+        }
+        return u;
+    }, [selectedIds, teammateNameDisplay, onRemove]);
+
+    const marginBottom = useMemo(() => {
```

Comment:
Since this is a number, not really need to memoize, but since the operation is a bit complex, let's keep it like this so it is better compartmentalized.